docs/how-to/deploy_sdcore_user_plane_with_hugepages.md

```diff
@@ -0,0 +1,67 @@
+# Deploy SD-Core User Plane with HugePages
+
+This guide covers how to deploy the User Plane Function (UPF) with huge pages using 
+the `sdcore-user-plane` Juju bundle.
+
+## Requirements
+
+- A Kubernetes host which meets the following requirements: 
+  - CPU that supports AVX2, RDRAND and PDPE1GB instructions (Intel Haswell, AMD Excavator or equivalent)
+  - LoadBalancer with 1 available address for the UPF
+  - Multus CNI enabled
+- Juju >= 3.1/stable
+- A Juju controller bootstrapped onto the Kubernetes host
+- Juju model created, named `user-plane`
+
+## Enable HugePages on the host
+As `root` user, edit the bootloader configuration of the OS (`/etc/default/grub`) appending the following command to the
+kernel command-line parameters (`GRUB_CMDLINE_LINUX_DEFAULT`):
+```console
+default_hugepagesz=1G
+```
+Then update the bootloader:
+```shell
+update-grub
+```
+
+Allocate 2 HugePages of 1Gi, for a total of 2Gi HugePages, and make it persistent:
+```shell
+echo 2 | tee /proc/sys/vm/nr_hugepages
```

Comment:
This line is not required, as we set it persistently right after and tell the user to reboot.